pycbc/inference/sampler_kombine.py

```diff
@@ -0,0 +1,194 @@
+        kombine.clustered_kde.TransdimensionalKDE proposal distribution. In
+        this mode a masked array with samples in each of the possible sets of
+        dimensions must be given for the initial ensemble distribution.
+    processes : {None, int}
+        Number of processes to use with multiprocessing. If None, all available
+        cores are used.
+    min_burn_in : {None, int}
+        Set the minimum number of burn in iterations to use. If None,
+        `burn_in_iterations` will be initialized to `0`.
+    """
+    name = "kombine"
+
+    def __init__(self, likelihood_evaluator, nwalkers, transd=False,
+            processes=None, min_burn_in=None):
+        try:
+            import kombine
+        except ImportError:
+            raise ImportError("kombine is not installed.")
+
+        # construct sampler for use in KombineSampler
+        ndim = len(likelihood_evaluator.waveform_generator.variable_args)
+        sampler = kombine.Sampler(nwalkers, ndim, likelihood_evaluator,
+                                          transd=transd, processes=processes)
+        # initialize
+        super(KombineSampler, self).__init__(sampler, likelihood_evaluator,
+            min_burn_in=min_burn_in)
+        self._nwalkers = nwalkers
+
+    @classmethod
+    def from_cli(cls, opts, likelihood_evaluator):
+        """Create an instance of this sampler from the given command-line
+        options.
+
+        Parameters
+        ----------
+        opts : ArgumentParser options
+            The options to parse.
+        likelihood_evaluator : LikelihoodEvaluator
+            The likelihood evaluator to use with the sampler.
+
+        Returns
+        -------
+        KombineSampler
+            A kombine sampler initialized based on the given arguments.
+        """
+        return cls(likelihood_evaluator, opts.nwalkers,
+                   processes=opts.nprocesses, min_burn_in=opts.min_burn_in)
+
+    def run(self, niterations, **kwargs):
+        """Advance the sampler for a number of samples.
+
+        Parameters
+        ----------
+        niterations : int
+            Number of samples to get from sampler.
+
+        Returns
+        -------
+        p : numpy.array
+            An array of current walker positions with shape (nwalkers, ndim).
+        lnpost : numpy.array
+            The list of log posterior probabilities for the walkers at
+            positions p, with shape (nwalkers, ndim).
+        lnprop : numpy.array
+            The list of log proposal densities for the walkers at positions p,
+            with shape (nwalkers, ndim).
+        """
+        blob0 = None
+        if self.burn_in_iterations == 0:
+            # no burn in, use the initial positions
+            p0 = self.p0
+            if self.likelihood_evaluator.return_meta:
+                blob0 = [self.likelihood_evaluator(p0[wi,:])[1]
+                    for wi in range(self.nwalkers)]
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->